### ISSUE TITLE
Trailing comma is forced when closing paren is on same line

### ISSUE THREAD

**Issue by ptarjan** [rubocop-hq/rubocop#2770]
This shouldn't need a trailing comma:

```
foo({
  bar: 1,
  baz: 2,
}, 3)
```

but if is being force to:

```
foo({
  bar: 1,
  baz: 2,
}, 3,)
```

**Comment by bbatsov** [comment]
Which style are you using?

**Comment by ptarjan** [comment]
@bbatsov The first style. I've enabled `TrailingCommas` and like it mostly except for the few extraneous commas I reported bugs about.

**Comment by alexdowad** [comment]
We have played with the existing styles for the `TrailingComma` cops before and got a whole bunch of blowback from people who didn't like the unexpected changes. I think @ptarjan's request is fairly reasonable, but wonder if it should be a new style, rather than messing with the existing ones...

Other contributors?

**Comment by jonas054** [comment]
If your configuration is
```yaml
Style/TrailingCommaInArguments:
  # If `comma`, the cop requires a comma after the last argument, but only for
  # parenthesized method calls where each argument is on its own line.
  # If `consistent_comma`, the cop requires a comma after the last argument,
  # for all parenthesized method calls with arguments.
  EnforcedStyleForMultiline: comma
  SupportedStyles:
    - comma
    - consistent_comma
    - no_comma
```
as the comment says, the cop accepts your first example and rejects the second. Are you sure you're not using the `consistent_comma` style?

**Comment by mikegee** [comment]
> Are you sure you're not using the `consistent_comma` style?

@ptarjan's report indicates using `consistent_comma`.

Given this file:

```ruby
foo({
      bar: 1,
      baz: 2,
    }, 3)
```

Here are the offenses from the available styles:

### EnforcedStyleForMultiline: comma (the default)

```
test/test.rb:3:13: C: Style/TrailingCommaInHashLiteral: Avoid comma after the last item of a hash.
      baz: 2,
            ^

1 file inspected, 1 offense detected
```

### EnforcedStyleForMultiline: consistent_comma

```
test/test.rb:3:13: C: Style/TrailingCommaInHashLiteral: Avoid comma after the last item of a hash.
      baz: 2,
            ^
test/test.rb:4:8: C: Style/TrailingCommaInArguments: Put a comma after the last parameter of a multiline method call.
    }, 3)
       ^

1 file inspected, 2 offenses detected
```

### EnforcedStyleForMultiline: no_comma

```
test/test.rb:3:13: C: Style/TrailingCommaInHashLiteral: Avoid comma after the last item of a hash.
      baz: 2,
            ^

1 file inspected, 1 offense detected
```

It is unfortunate that the example file has no configuration without offenses, but is it worth supporting another style? I suppose since this issue has been open for so long without additional user complaints indicates it isn't a huge problem.

**Comment by bbatsov** [comment]
There hasn't been much activity on this ticket and our Core Team is spread too thin on so many tasks, so I'll just close it for the sake of having a cleaner lists of tasks to focus on. We'd gladly review a PR, but it's unlikely that'd we're going to tackle this ourselves in the foreseeable future.
